shell/components/Resource/Detail/PercentageBar.vue

```diff
@@ -0,0 +1,45 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+
+export interface Props {
+  percent: number;
+  percentageColor?: string;
+  baseColor?: string;
+}
+
+const props = withDefaults(
+  defineProps<Props>(),
+  {
+    baseColor:       'var(--progress-bg)',
+    percentageColor: 'var(--primary)'
+  }
+);
+
+const baseStyle = computed(() => ({ backgroundColor: props.baseColor }));
+const usedStyle = computed(() => ({
+  backgroundColor: props.percentageColor,
+  width:           `${ props.percent }%`
+}));
+</script>
+
+<template>
+  <div
+    class="percentage-bar"
+    :style="baseStyle"
+  >
+    <div
+      class="used"
+      :style="usedStyle"
+    >
+&nbsp;
```

Comment:
nit: can we replace `&nbsp` usage with css?

Reply:
done.